cmd/nanotube/listen.go

```diff
@@ -124,25 +124,28 @@ loop:
 	close(queue)
 }
 
-func listenUDP(conn net.Conn, queue chan string, stop <-chan struct{},
-	UDPbBufSize int, lg *zap.Logger, ms *metrics.Prom) {
+func listenUDP(conn net.Conn, queue chan string, stop <-chan struct{}, lg *zap.Logger, ms *metrics.Prom) {
 	defer func() {
 		cerr := conn.Close()
 		if cerr != nil {
 			lg.Error("closing the incoming connection", zap.Error(cerr))
 		}
 	}()
 
-	r := bufio.NewReaderSize(conn, UDPbBufSize)
+	buf := make([]byte, 64*1024) // 64k is the max UDP datagram size
 	for {
-		line, err := r.ReadBytes('\n')
-		if err != nil && err != io.EOF {
-			lg.Error("error reading a line from UDP connection", zap.Error(err))
+		nRead, err := conn.Read(buf)
+		if err != nil {
+			lg.Error("error reading UDP datagram", zap.Error(err))
 			continue
 		}
 
+		lines := bytes.Split(buf[:nRead], []byte{'\n'})
+
 		// TODO (grzkv): string -> []bytes
-		sendToMainQ(string(line), queue, ms)
+		for i := 0; i < len(lines)-1; i++ {
+			sendToMainQ(string(lines[i]), queue, ms)
```

Comment:
When we get to implement string -> []bytes we will have to be careful about race conditions on "buf".
Split() on line 143 returns collections of slices in the same array(line 135).
If we send lines[] to another goroutine, we will have this goroutine updating buf(line 137) while other goroutine is reading it.
Added suggestion for comment so we remember about this in the future.

Reply:
I am aware. Thanks. Added comments.